project/scenes/Main.gd

```diff
@@ -28,17 +45,41 @@ func new_game():
 	$Monkey.velocity = Vector2i(0, 0)
 	$Camera2D.position = CAM_START_POS
 	$ground.position = Vector2i(0, 0)
+	get_tree().paused = false
+	difficulty = 0
+	
+	# reset obstacles
+	for obs in obstacles:
+		obs.queue_free()
+	obstacles.clear()
+	
+	$GameOver.hide()
 	
 
 # Called every frame. 'delta' is the elapsed time since the previous frame.
 func _process(delta):
 	if player_on_camera:
-		speed = START_SPEED + score / 5
+		speed = START_SPEED + score / 10
+		
+		if speed > MAX_SPEED:
+			speed = MAX_SPEED
```

Comment:
doesn't Godot have a function for capping variables like this? or just a `min()` function to handle this